### PR TITLE
Release 7.3.1 version of che-plugin-registry

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?

- Cherry-picking all commits from master
- Bumping version to 7.3.1
- Adding 7.3.1 versions of 'che-theia' and 'che-machine-exec' and updating `latest.txt` of those plugins
- Setting '7.3.1' version for container runtime images of the che remote plugins via `find . -name '*.yaml' -exec sed -i -e 's/:7.3.0/:7.3.1/g' {} \;`

### What issue does this PR reffer to?
https://github.com/eclipse/che/issues/14947

### CODE REVIEW

File: v3/plugins/redhat/vscode-openshift-connector/0.1.0/meta.yaml [hunk 1]
```diff
@@ -12,8 +12,10 @@ category: Other
 firstPublicationDate: "2019-10-08"
 spec:
   containers:
-    - image: "docker.io/eclipse/che-remote-plugin-openshift-connector-0.1.0:7.3.0"
+    - image: "docker.io/eclipse/che-remote-plugin-openshift-connector-0.1.0:7.3.1"
       name: "vscode-openshift-connector"
       memoryLimit: "512Mi"
   extensions:
+    - https://github.com/Azure/vscode-kubernetes-tools/releases/download/1.0.0/vscode-kubernetes-tools-1.0.0.vsix
```
[the remark this answers is not in the document]
Comment (nickboldt) [reply v3/plugins/redhat/vscode-openshift-connector/0.1.0/meta.yaml]: ah, it's https://github.com/eclipse/che-plugin-registry/commit/097a56f8b01d14af7fd6ffa8aff376fa91cc6834

[the remark this answers is not in the document]
Comment (nickboldt) [reply v3/plugins/redhat/vscode-openshift-connector/0.1.0/meta.yaml]: so if https://github.com/eclipse/che-plugin-registry/commit/097a56f8b01d14af7fd6ffa8aff376fa91cc6834#diff-525d44ea1d1f41ad8ca64916cc13850aR70 is for /next/ does that mean it's not applicable to 7.3.1 ?

[the remark this answers is not in the document]
Comment (ibuziuk) [reply v3/plugins/redhat/vscode-openshift-connector/0.1.0/meta.yaml]: @nickboldt my understanding that this PR https://github.com/eclipse/che-plugin-registry/pull/233  is addressing https://github.com/eclipse/che/issues/14797 at some point, so it must be in 7.3.1
@AndrienkoAleksandr @evidolob could you comment?

File: v3/plugins/eclipse/che-theia/7.3.1/meta.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 publisher: eclipse
 name: che-theia
-version: latest
+version: 7.3.1
 type: Che Editor
 displayName: theia-ide
-title: Eclipse Theia for Eclipse Che
-description: Eclipse Theia
+title: Eclipse Theia development version.
+description: Eclipse Theia, get the latest release each day.
```
Comment (amisevsk) [inline v3/plugins/eclipse/che-theia/7.3.1/meta.yaml]: This title and description seems to be inaccurate, no? The version will be fixed here.

File: v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.0.4/meta.yaml [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 repository: https://github.com/Azure/vscode-kubernetes-tools
 category: Other
-firstPublicationDate: '2019-05-15'
+firstPublicationDate: "2019-10-16"
 spec:
   containers:
-  - image: "docker.io/eclipse/che-remote-plugin-kubernetes-tooling-1.0.0:7.3.0"
-    name: vscode-kubernetes-tools
-    memoryLimit: "1G"
+    - image: "docker.io/eclipse/che-remote-plugin-kubernetes-tooling-1.0.4:next"
```
Comment (amisevsk) [inline v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.0.4/meta.yaml]: Should be tagged `7.3.1`? This is the "latest" version.

**Comment by ibuziuk** [comment]
> Also, I don't know our policy on adding new plugins to a bugfix release, but the following plugins are present in this PR but not in 7.3.0:
> 
> /v3/plugins/pizzafactory/xtend-lang
> ./v3/plugins/pizzafactory/vscode-libra-move
> ./v3/plugins/pizzafactory/xtext-lang

@nickboldt @l0rd I'm also not sure if we should keep it or remove of `7.3.1`

File: v3/plugins/eclipse/che-theia/7.3.1/meta.yaml (hunk 2 above)
Comment (ibuziuk) [reply v3/plugins/eclipse/che-theia/7.3.1/meta.yaml]: good catch

File: v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.0.4/meta.yaml (hunk 3 above)
Comment (ibuziuk) [reply v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.0.4/meta.yaml]: @amisevsk good catch, I assume if we would like to keep `pizzafactory` plugins in `7.3.1` the container runtime image should be also updated there

**Comment by l0rd** [comment]
@ibuziuk we should probably remove them. If that's complicated let's discuss it.

File: v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.0.4/meta.yaml (hunk 3 above)
Comment (amisevsk) [reply v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.0.4/meta.yaml]: :+1: @monaka any issues with this?

**Comment by amisevsk** [comment]
There should be no harm in just `rm -rf`ing those directories; I agree we should wait until 7.4.0 to introduce them.

**Comment by ibuziuk** [comment]
@amisevsk @nickboldt @l0rd PR has been updated - addressed Angel's review comments + removed `pizzafactory` plugins

**Pull request merged**